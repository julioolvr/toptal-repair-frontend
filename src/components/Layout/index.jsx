import React from 'react';
import PropTypes from 'prop-types';
import g from 'glamorous';

import UserNavbar from '../UserNavbar';

function Layout({ isUserAuthenticated, children }) {
  return (
    <g.Div
      height="100vh"
      padding="3em"
      boxSizing="border-box"
      backgroundColor="#99ddff"
      color="#333"
    >
      {isUserAuthenticated ? <UserNavbar /> : null}
      {children}
    </g.Div>
  );
}

Layout.propTypes = {
  isUserAuthenticated: PropTypes.bool.isRequired,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
};

export default Layout;
