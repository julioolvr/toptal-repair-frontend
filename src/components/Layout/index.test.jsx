import React from 'react';
import { shallow } from 'enzyme';

import Layout from '.';

it('renders a layout without a user', () => {
  const wrapper = shallow(
    <Layout isUserAuthenticated={false}>
      <div>Content</div>
    </Layout>,
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders a layout with a user', () => {
  const wrapper = shallow(
    <Layout isUserAuthenticated>
      <div>Content</div>
    </Layout>,
  );
  expect(wrapper).toMatchSnapshot();
});
