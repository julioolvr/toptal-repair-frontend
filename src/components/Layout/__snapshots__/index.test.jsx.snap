// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a layout with a user 1`] = `
<glamorous.Div
  backgroundColor="#99ddff"
  boxSizing="border-box"
  color="#333"
  height="100vh"
  padding="3em"
>
  <Connect(UserNavbar) />
  <div>
    Content
  </div>
</glamorous.Div>
`;

exports[`renders a layout without a user 1`] = `
<glamorous.Div
  backgroundColor="#99ddff"
  boxSizing="border-box"
  color="#333"
  height="100vh"
  padding="3em"
>
  <div>
    Content
  </div>
</glamorous.Div>
`;
